shared/src/main/scala/implicits.scala

```diff
@@ -16,29 +16,71 @@
 
 package fetch
 
-import cats.MonadError
+
+import java.util.{Timer, TimerTask}
+import java.util.concurrent.TimeoutException
+import scala.concurrent.duration._
 import cats.instances.future._
 import scala.concurrent.{ExecutionContext, Future, Promise}
 
 object implicits {
 
-  implicit def fetchFutureFetchMonadError(
-      implicit ec: ExecutionContext
-  ): FetchMonadError[Future] =
+  // Shared Timer object to schedule timeouts
+  val timer: Timer = new Timer()
+
+  implicit def fetchFutureFetchMonadError(implicit ec: ExecutionContext): FetchMonadError[Future] =
     new FetchMonadError.FromMonadError[Future] {
       override def runQuery[A](j: Query[A]): Future[A] = j match {
-        case Sync(e) => Future(e.value)
-        case Async(ac, timeout) => {
+
+        case Sync(e) =>
+          Future.successful({e.value})
```

Comment:
Evaluating `e` could potentially throw an exception and those aren't caught by `Future.successful`.